### PR TITLE
Added seed phrases to KeyNMF

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
You can now add a `seed_phrase` to a KeyNMF model, which essentially indicates the aspect, from which the model has to examine documents.

```python
from sklearn.datasets import fetch_20newsgroups

from turftopic import KeyNMF

corpus = fetch_20newsgroups(
    subset="all",
    remove=("headers", "footers", "quotes"),
).data

model = KeyNMF(5, seed_phrase="Is homosexuality moral?")
model.fit(corpus)

model.print_topics()
```

| Topic ID | Highest Ranking |
| - | - |
| 0 | homosexuality, homosexual, immoral, sodom, heterosexual, sexual, fornication, christians, verses, sex |
| 1 | morality, moral, immoral, morals, objective, morally, society, animals, behavior, natural |
| 2 | christians, christian, christianity, religion, bible, god, church, religious, faith, beliefs |
| 3 | homosexual, homosexuals, heterosexual, gay, sexual, sex, heterosexuals, straight, men, sexuality |
| 4 | sin, sins, god, sinner, sinful, condemnation, sinned, scripture, punishment, sinners |

### TODO:
 - [x] Add documentation
 - [x] Release new version

### CODE REVIEW

**Comment by x-tabdeveloping** [comment]
@KennethEnevoldsen can I has review?

File: docs/KeyNMF.md [hunk 1]
```diff
@@ -94,70 +104,89 @@ can be explained.
...
+    model.fit(corpus, embeddings=embeddings)
+    ```
+=== "Pre-computed keyword matrix"
+    ```python
+    keyword_matrix = model.extract_keywords(corpus)
+    model.fit(None, keywords=keyword_matrix)
+    ```
 
-### Asymmetric and Instruction-tuned Embedding Models
+## Seeded Topic Modeling
```
Comment (KennethEnevoldsen) [inline docs/KeyNMF.md:129]: Do you need to refer to the general documentation of seeded topic modeling here (seems like a lot of duplication). 

Might it not be better to create a table like "supported types of topic modelling" and then in the "Seeded Topic Modelling" section add "Models which support seeded topic modelling".

File: docs/seeded.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Seeded Topic Modeling
+
+When investigating a set of documents, you might already have an idea about what aspects you would like to explore.
+Some models are able to account for this by taking seed phrases or words.
+This is currently only possible with KeyNMF in Turftopic, but will likely be extended in the future.
+
+In [KeyNMF](../keynmf.md), you can describe the aspect, from which you want to investigate your corpus, using a free-text seed-phrase,
```
Comment (KennethEnevoldsen) [inline docs/seeded.md:7]: See the comment above as well.

Would probably write this more simply and then put this in a Tab section called KeyNMF (that way it is easy to see that the only one supported in KeyNMF, but also that there could be others in the future.

File: turftopic/models/_keynmf.py [hunk 3]
```diff
@@ -120,6 +120,8 @@ def batch_extract_keywords(
         self,
         documents: list[str],
         embeddings: Optional[np.ndarray] = None,
+        seed_embedding: Optional[np.ndarray] = None,
+        fitting: bool = True,
```
Comment (KennethEnevoldsen) [inline turftopic/models/_keynmf.py:124]: What fitting do?

**Review by KennethEnevoldsen — COMMENTED** [review]
Looking great a few ideas to restructure the docs - nothing holding back this PR though

File: docs/KeyNMF.md [hunk 4]
```diff
@@ -354,46 +424,49 @@ for batch in batched(zip(corpus, timestamps)):
     model.partial_fit_dynamic(text_batch, timestamps=ts_batch, bins=bins)
 ```
 
-### Hierarchical Topic Modeling
+## Asymmetric and Instruction-tuned Embedding Models
```
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply docs/KeyNMF.md:427]: They are also in the general section, but I put them here as well, since they are relevant enough for KeyNMF

File: docs/KeyNMF.md (hunk 1 above)
Comment (x-tabdeveloping) [reply docs/KeyNMF.md:129]: I don't mind duplication. I think the most important thing by far is that people find relevant information about the thing they click on. This is not a SQL database, I think it's fine to repeat things as long as it makes the docs easier to navigate.

File: docs/KeyNMF.md [hunk 5]
```diff
@@ -94,70 +104,89 @@ can be explained.
...
 
-### Asymmetric and Instruction-tuned Embedding Models
+## Seeded Topic Modeling
 
-Some embedding models can be used together with prompting, or encode queries and passages differently.
-This is important for KeyNMF, as it is explicitly based on keyword retrieval, and its performance can be substantially enhanced by using asymmetric or prompted embeddings.
-Microsoft's E5 models are, for instance, all prompted by default, and it would be detrimental to performance not to do so yourself.
+When investigating a set of documents, you might already have an idea about what aspects you would like to explore.
+In KeyNMF, you can describe this aspect, from which you want to investigate your corpus, using a free-text seed-phrase,
+which will then be used to only extract topics, which are relevant to your research question.
```
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply docs/KeyNMF.md:133]: Not in the form of seed phrases, no. You can do something similar in BERTopic with seed words either for a predefined topic or with c-tf-idf, but it's not exactly the same concept.

File: turftopic/models/_keynmf.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply turftopic/models/_keynmf.py:124]: Ahh, would probably add a docstring to it
